src/ITCC.YandexSpeeckKitClient/SpeechKitClient.cs

```diff
@@ -0,0 +1,254 @@
+﻿// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
+// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using ITCC.YandexSpeeckKitClient.Enums;
+using ITCC.YandexSpeeckKitClient.Extensions;
+using ITCC.YandexSpeeckKitClient.Models;
+using ITCC.YandexSpeeckKitClient.Utils;
+
+namespace ITCC.YandexSpeeckKitClient
+{
+    /// <summary>
+    /// Yandex SpeechKit Cloud API client.
+    /// </summary>
+    public class SpeechKitClient : IDisposable
+    {
+        #region Private fields
+
+        private readonly string _apiKey;
+        private readonly string _applicationName;
+        private readonly Guid _userId;
+        private readonly string _device;
+        private readonly HttpClient _httpClient = new HttpClient();
+        private readonly int _timeout;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Create new Yandex SpeechKit Cloud API client.
+        /// </summary>
+        /// <param name="apiKey">The API key.</param>
+        /// <param name="applicationName">Name of the client application.</param>
+        /// <param name="userId">User's identifier.</param>
+        /// <param name="device">The type of device running the client application.</param>
+        /// <param name="timeout">Data streaming operation's timeout.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public SpeechKitClient(string apiKey, string applicationName, Guid userId, string device, TimeSpan timeout)
+            : this(apiKey, applicationName, userId, device, (int) timeout.TotalMilliseconds)
+        {
+        }
+
+        /// <summary>
+        /// Create new Yandex SpeechKit Cloud API client.
+        /// </summary>
+        /// <param name="apiKey">The API key.</param>
+        /// <param name="applicationName">Name of the client application.</param>
+        /// <param name="userId">User's identifier.</param>
+        /// <param name="device">The type of device running the client application.</param>
+        /// <param name="timeout">Data streaming operation's timeout in milliseconds.</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public SpeechKitClient(string apiKey, string applicationName, Guid userId, string device, int timeout = Configuration.DefaultTimeout)
+        {
+            _apiKey = string.IsNullOrWhiteSpace(apiKey) 
+                ? throw new ArgumentNullException(nameof(apiKey)) 
```

Comment:
Would be better to split cases... Don't think passing empty string should result in `ArgumentNullException`